components/core/src/clp_s/Utils.hpp

```diff
@@ -264,10 +267,87 @@ class Span {
 
     T& operator[](size_t i) { return m_begin[i]; }
 
+    Span<T> sub_span(size_t start, size_t size) { return {m_begin + start, size}; }
+
 private:
     T* m_begin;
     size_t m_size{};
 };
 }  // namespace clp_s
 
+template <typename T>
+class UnalignedSpan {
```

Comment:
Yeah, the main difference is that the underlying buffer isn't guaranteed to be aligned. This doesn't really make a difference on newer x86 machines (besides some cache effects), but does matter for older x86.

Reply:
For our use case the main difference is really just that we need to do that memcpy when reading to avoid getting into c++ UB. Any modern compiler should turn that memcpy into a normal load.

Reply:
Can you come up with an example where alignment is an issue between different architectures since we only use fundamental types?

Reply:
Also good to know for compound types even if we don't use that.

Reply:
I guess the main concrete overhead is that sometimes our reads will cross a cache-line boundary and incur some extra overhead. There's also some RISC architectures like MIPS that just don't support unaligned accesses at all.